### PR TITLE
Add mp11

### PR DESCRIPTION
**Pull request by brunocodutra**

### CODE REVIEW

**Comment by brunocodutra** [comment]
@odinthenerd is that all you had so far?

File: benchmark/dependencies/mp11.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright Odin Holmes 2017
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+
+if (METABENCH_MP11
+    AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+             CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
```
Comment (pdimov) [inline benchmark/dependencies/mp11.cmake]: 4.7 and 4.8 should work too.

File: benchmark/dependencies/mp11.cmake [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
+             CMAKE_CXX_COMPILER_VERSION VERSION_LESS "3.6")
```
Comment (pdimov) [inline benchmark/dependencies/mp11.cmake]: Here 3.5 works, 3.4 should as well.

Comment (brunocodutra) [inline benchmark/dependencies/mp11.cmake]: ok

**Comment by pdimov** [comment]
Fails due to missing `<boost/config.hpp>`. :-(

**Comment by brunocodutra** [comment]
Oh good catch, I'll fix that too

**Comment by brunocodutra** [comment]
@ldionne this should be ready to merge if CI passes
@pdimov please shout out if you see anything odd

**Comment by pdimov** [comment]
Seems insert/erase break the g++ builds. I just changed `mp_take` to be less slow though, so they should work better now. If not, let's limit them to something like 200-300.

**Comment by brunocodutra** [comment]
>  so they should work better now
they did, tests passed, thanks

**Pull request merged**